build-system/tasks/storybook/index.js

```diff
@@ -42,14 +48,60 @@ function runStorybook(mode) {
   );
 }
 
+/**
+ * @param {boolean} port
+ * @return {!Promise<boolean>}
+ */
+const isPortInUse = (port) =>
+  new Promise((resolve) => {
+    const server = createServer();
+    server.once('error', (err) => {
+      if (err.code === 'EADDRINUSE') {
+        resolve(true);
+      }
+    });
+    server.once('listening', function () {
+      resolve(false);
+      server.close();
+    });
+    server.listen(port);
+  });
+
```

Comment:
Infra folk: should this be moved somewhere?

Reply:
I think it would be useful to move this logic, and possibly part of the related warning messages, into the server code itself. This would probably be useful to anyone working with the server who accidentally is running multiple copies and eating up ports (I know I've done this)

Reply:
Yeah, maybe it's better if this just works consistently like `serve`. I'll undo `serve` changes, since there's no need for a double `gulp` anymore bc `argv.port` was renamed.